File: extract.ts
import fs from 'node:fs';
import pdf from 'pdf-parse';

async function extract() {
    try {
        const fileBuffer = fs.readFileSync('Web Dev - Question Paper.pdf');
        const data = await pdf(fileBuffer);
        fs.writeFileSync('pdf-text.txt', data.text);
        console.log('Successfully wrote PDF to pdf-text.txt');
    } catch (e) {
        console.error('Error reading PDF:', e);
    }
}

extract();

File: extract.py
import pypdf

try:
    reader = pypdf.PdfReader('Web Dev - Question Paper.pdf')
    text = '\n'.join(page.extract_text() for page in reader.pages)
    with open('pdf-text.txt', 'w', encoding='utf-8') as f:
        f.write(text)
    print("PDF extraction complete.")
except Exception as e:
    print(f"Error: {e}")
